### DISCUSSION TITLE
AAXC support?

### DISCUSSION THREAD

**Discussion by nl255** [audiamus/AaxAudioConverter#124] [Ideas]
Would it be possible to add at least basic support for .aaxc files even if it means you need to provide the .voucher file yourself after having used audible-cli?  Or at least some way to convert .aaxc files to the .aax/json format that is expected?

**Comment by audiamus** [comment]
Did you have a look at [Book Lib Connect](https://github.com/audiamus/BookLibConnect), the new companion to AAX Audio Converter?

**Reply by refskegg** [reply]
_The English translation can be found below._
Danke dir erstmal für deine Arbeit, ich bin auf dein Programm in einem deutschsprachigen Forum gestoßen, Book Lib Connect und der AAX Audio Converter gehen reibungslos. 
Ich hab absolut keine Ahnung von der Technik dahinter (vorallem bei Book Lib Connect, für mich ist das Magie, wie plötzlich das Hörbuch da ist), aber mir ist letztens auch die Idee gekommen die Hörbücher von Audible übers Handy auf SD-Karte zu ziehen und dann zu konvertieren. Die sind aber im AAXC Format. Keine Ahnung wo da der Unterschied ist, man kann sie aber nicht mit dem AAX Audio Converter konvertieren. 
Ich kann verstehen, dass wenn Book Lib Connect funktioniert du vielleicht wenig Bock drauf hast AAXC Konvertierung zu implementieren, aber mich interessiert es auch einfach von der technischen Seite. Wo liegt denn der Unterschied zwischen AAX und AAXC und warum liegt es per Handydownload in AAXC vor und per Book Lib Connect in AAX? 

_English:_ 
Thank you for your work, I came across your program in a German-language forum, and Book Lib Connect and the AAX Audio Converter work smoothly. I have absolutely no idea about the technology behind it (especially with Book Lib Connect, for me, it's like magic how the audiobook suddenly appears), but I also recently had the idea to download audiobooks from Audible on my phone, transfer them to an SD card, and then convert them. However, they are in AAXC format. I have no idea what the difference is, but you can't convert them with the AAX Audio Converter.

I can understand that if Book Lib Connect is working, you might not be too eager to implement AAXC conversion, but I'm also interested in it from a technical perspective. What is the difference between AAX and AAXC, and why is it in AAXC format when downloaded on a mobile device and in AAX format through Book Lib Connect?

**Reply by audiamus** [reply]
Book Lib Connect lädt AAXC herunter und macht daraus M4B. Die exportierte AAX-Datei ist lediglich die umbenannte M4B-Datei, um kompatibel mit AAX Audio Converter zu werden, segelt also ein wenig unter falscher Flagge. 

Bei AAX gab/gibt es einen einheitlichen Schlüssel für alle Hörbücher zu einem Konto. Bei AAXC wird der Schlüssel (in einem Voucher enthalten) individuell für jede Kombination aus Kunde, Buch und Gerät ausgestellt. Und ein solches Gerät, in diesem Fall ein iPhone, wird von Book Lib Connect emuliert.

Das ganze ist recht komplex. Hier zwei weiterführende Diskussionsbeiträge:
https://github.com/audiamus/BookLibConnect/discussions/100#discussioncomment-5542068
https://github.com/audiamus/BookLibConnect/discussions/92#discussioncomment-4968738
